RESTEasy Singleton resource with constructor injection failing with RESTEASY003190

## Describe the bug
When using constructor injection in a non annotated resource ([reproducer link](https://github.com/chris-asl/quarkus-mcve/commit/2234069170ec55e341388ad5e12445269eb320a3#diff-c1e351af7879a35f1479bc0907392e13720790b0c7ccabcd4acbf4da3ef175bcR11-R19)), e.g.

```java
@Path("/hello")
public class GreetingResource {

    GreetingService greetingService;

    @Inject
    public GreetingResource(GreetingService greetingService) {
        this.greetingService = greetingService;
    }
```

the program spins up successfully.

Given the default resource scope, as defined by [`quarkus.resteasy.singleton-resources`](https://quarkus.io/guides/all-config#quarkus-resteasy-server-common_quarkus.resteasy.singleton-resources) is `Singleton`, why do I get an error if I annotate the class with `@Singleton`, as [I did here](https://github.com/chris-asl/quarkus-mcve/commit/72a40a7f2f7f25bba2e50a634af1a333145806d8)?

The program fails upon startup with:
```
...
Caused by: java.lang.RuntimeException: RESTEASY003190: Could not find constructor for class: org.acme.getting.started.GreetingResource
	at org.jboss.resteasy.spi.metadata.ResourceBuilder.getConstructor(ResourceBuilder.java:852)
...
```

It seems there's a difference with explicitly setting the scope to `Singleton` with it being set-up "automatically".  
If I am missing something, a pointer towards the docs would be more than welcome.

I don't know if this is related to https://github.com/quarkusio/quarkus/issues/6722 or https://github.com/quarkusio/quarkus/issues/7495.

### Expected behavior
As in both cases the scope effectively is `Singleton`, shouldn't the behaviour be the same?

### Actual behavior
Annotating a RESTEasy Resource with `@Singleton`, I get the aforementioned error.

## To Reproduce

Link to a small reproducer (preferably a Maven project if the issue is not Gradle-specific).

Or attach an archive containing the reproducer to the issue.

Steps to reproduce the behavior:
1. `git clone git@github.com:chris-asl/quarkus-mcve.git`
2. `./mvnw quarkus:dev`

### Configuration
Defaults

## Environment:

### Output of `uname -a` or `ver`

`Linux homer 5.8.0-48-generic #54~20.04.1-Ubuntu SMP Sat Mar 20 13:40:25 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux`

### Output of `java -version`

```
openjdk version "11.0.10" 2021-01-19
OpenJDK Runtime Environment (build 11.0.10+9-Ubuntu-0ubuntu1.20.04)
OpenJDK 64-Bit Server VM (build 11.0.10+9-Ubuntu-0ubuntu1.20.04, mixed mode, sharing)
```

### Quarkus version or git rev
`1.13.1.Final` (also tested with `1.12.2.Final`)

### Build tool

```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /home/chris/.m2/wrapper/dists/apache-maven-3.6.3-bin/1iopthnavndlasol9gbrbg6bf2/apache-maven-3.6.3
Java version: 11.0.10, vendor: Ubuntu, runtime: /usr/lib/jvm/java-11-openjdk-amd64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.8.0-48-generic", arch: "amd64", family: "unix"
```

FYI @el10686

--- a/extensions/resteasy-classic/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java
+++ b/extensions/resteasy-classic/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java
@@ -459,6 +459,7 @@ public class ResteasyServerCommonProcessor {
         prefixes.add(new AllowedJaxRsAnnotationPrefixBuildItem("javax.annotation.security"));
         prefixes.add(new AllowedJaxRsAnnotationPrefixBuildItem("jakarta.annotation.security"));
         prefixes.add(new AllowedJaxRsAnnotationPrefixBuildItem("java.lang"));
+        prefixes.add(new AllowedJaxRsAnnotationPrefixBuildItem("javax.inject"));
         return prefixes;
     }
 

--- a/extensions/resteasy-classic/resteasy/deployment/src/test/java/io/quarkus/resteasy/test/ConstructorInjectionResourceTestCase.java
+++ b/extensions/resteasy-classic/resteasy/deployment/src/test/java/io/quarkus/resteasy/test/ConstructorInjectionResourceTestCase.java
@@ -14,10 +14,16 @@ public class ConstructorInjectionResourceTestCase {
     @RegisterExtension
     static QuarkusUnitTest runner = new QuarkusUnitTest()
             .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
-                    .addClasses(ConstructorInjectionResource.class, Service.class));
+                    .addClasses(ConstructorInjectionResource.class, SingletonConstructorInjectionResource.class,
+                            Service.class));
 
     @Test
     public void testConstructorInjectionResource() {
         RestAssured.when().get("/ctor").then().body(Matchers.is("service"));
+    }
+
+    @Test
+    public void testSingletonConstructorInjectionResource() {
+        RestAssured.when().get("/ctor-single").then().body(Matchers.is("service"));
     }
 }

--- a/extensions/resteasy-classic/resteasy/deployment/src/test/java/io/quarkus/resteasy/test/SingletonConstructorInjectionResource.java
+++ b/extensions/resteasy-classic/resteasy/deployment/src/test/java/io/quarkus/resteasy/test/SingletonConstructorInjectionResource.java
@@ -1,0 +1,21 @@
+package io.quarkus.resteasy.test;
+
+import javax.inject.Singleton;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+
+@Singleton
+@Path("/ctor-single")
+public class SingletonConstructorInjectionResource {
+
+    final Service service;
+
+    public SingletonConstructorInjectionResource(Service service) {
+        this.service = service;
+    }
+
+    @GET
+    public String val() {
+        return service.execute();
+    }
+}

